plugins/module_utils/hmc_resource.py

```diff
@@ -805,3 +805,31 @@ def list_all_lpars_details(self, sys_name, filter=None):
     def getSystemNameFromMTMS(self, system_name):
         attr_dict = self.getManagedSystemDetails(system_name)
         return attr_dict.get('name')
+
+    def copyViosImage(self, sftp_server, sftp_user, sftp_password, image_name):
+        cpviosimgCmd = self.CMD['CPVIOSIMG'] +\
+            self.OPT['CPVIOSIMG']['-R']['SFTP'] +\
+            self.OPT['CPVIOSIMG']['-N'] + image_name +\
+            self.OPT['CPVIOSIMG']['-H'] + sftp_server +\
+            self.OPT['CPVIOSIMG']['-U'] + sftp_user +\
+            self.OPT['CPVIOSIMG']['-F'] + 'flash.iso' +\
+            self.OPT['CPVIOSIMG']['--PASSWD'] + sftp_password
+
+        self.hmcconn.execute(cpviosimgCmd)
+
+    def listViosImages(self, image_name=None):
+        if image_name:
+            lsviosimgCmd = self.CMD['LSVIOSIMG'] +\
+                '| grep -w ' +  image_name +\
+                ' || echo No results were found'
+        else:
+            lsviosimgCmd = self.CMD['LSVIOSIMG']
```

Comment:
if there are no images exist then this Line of code also returns as "No results were found" ?

Reply:
yes the cmd will return "No results were found" string